### ISSUE TITLE
FSE: Finalizing the name and menu item placement

### ISSUE THREAD

**Issue by aurooba** [WordPress/gutenberg#29630]
As we get close to releasing FSE into core, questions have come up **about what the feature will be officially called** within the WordPress Dashboard. 

This is also currently a blocker for the Documentation team ([see more details in this post on Make](https://make.wordpress.org/docs/2021/02/18/a-home-and-a-name-for-site-editor-documentation-full-site-editing-feature/)). The sooner a decision is reached on the name, the sooner we can spin up proper documentation for it.

@mtias also mentioned that a discussion is needed around **the placement of the menu item and how it interacts with the other items within Appearance sub menu**.

Related issue: #29031

**Comment by hedgefield** [comment]
This is a good point! "Site editor" and "Full site editing" is a nice name and it's what we've been calling it for a while, but as @mtias rightfully pointed out, is it the most descriptive name for what this thing is? I've seen "Design editor" as an option, though reading about the discussion of where to place it in the menu, I'm thinking "Appearance editor" might also be a good name?

Even more interesting is how to structure the appearance menu. I assume we'd want to keep some of the current menu items for backwards compatibility? Otherwise a lot of those items would become obsolete, right. Maybe something like this:

![Screenshot_1](https://user-images.githubusercontent.com/6073772/111476548-8edc9c00-872e-11eb-91d6-7ff5fe838511.png)

I removed a few smaller items that were basically just links to parts of the Customizer, and renamed "Theme Editor" to "Theme files editor" so it doesn't sound as much like it might overlap in functionality with the appearance editor.

It's too bad there's no precedent for using dividers in submenus, otherwise I could also see something like this to separate new from old a bit more

![Screenshot_2](https://user-images.githubusercontent.com/6073772/111476632-a025a880-872e-11eb-8b7e-5e0a451449f6.png)

**Comment by kellychoffman** [comment]
This is a great question. 

@jameskoster and I have been doing some explorations around this and have a prototype to share to illustrate an idea for how a user will be exposed to the new features that have been built as part of full site editing. During development, these were all contained within a single menu item called Full Site Editing. Going forward, as @hedgefield mentioned above, we need to think about how these features will integrate seamlessly into the current navigation menu structure so that it can be intuitive for both new and existing users. Doing so will also allow for pieces of it to be turned on and off. (Theme compatibility being one example)

Instead of a top level menu item called Full Site Editing, the main features within are pulled out and placed within Appearance menu:

![Appearance _ Themes](https://user-images.githubusercontent.com/5835847/113237792-596e9d00-925c-11eb-8b8b-3b6ae1fe9206.png)

[Try the prototype](https://www.figma.com/proto/LZiM9diybGa4WKM44zRcIq/Appearance-Menu?node-id=575%3A2302&viewport=132%2C325%2C0.047380756586790085&scaling=min-zoom).

[A 3.5 minute video walking through the prototype](https://us02web.zoom.us/rec/play/LZ12jVOG8fye34QzNHdbyAtRGu7WES3xRa51A6Y2LadMaRDEPMLj5xf5bMKsZlI_B-epyHiGbgk-ba3F.1kXtQwFMq9MBQu21?startTime=1617247805000).

> about what the feature will be officially called 

Back to the original question. This proposes that we call each feature what will be most familiar to the user. So, instead of "Full site editing" we refer to each one individually. ie, Styles, Templates, Header, Navigation (to start) and more that will be added as work continues on all other aspects a user needs to do to create, edit, and manage their site.

**Comment by hedgefield** [comment]
This is excellent! Thanks for the exploration 👏

**Comment by paaljoachim** [comment]
This looks very interesting! Nice work!

**NB! The above user interface is meant to be seen when a Block theme is activated!**

Discussions with James on Slack starting here: https://wordpress.slack.com/archives/C02S78ZAL/p1617267435267900

Some conclusions I have reached through discussions (before I have seen the above video and tested the prototype as I want to understand the concepts before testing.)

We have these links under Appearance:
Themes - Leads to Themes screen.
Styles - Leads to Global Styles.
Header (will show up if the theme has a defined Header FSE area.)
Footer  (If a Footer area has been defined a Footer link will also show up.)
Templates 
(Manage templates screen. Here one can manage the full site template that is in use. I do believe we might need a common manage screen for Template parts, Reusable blocks, Patterns etc. These can/should also be managed inside Gutenberg in a modal or in an isolated view. As one is working on the site one would like to stay within the Gutenberg screen without leaving it.)
Navigation

NB! It needs to be simple for an everyday user to add back the Customizer, Widgets screens and Theme Editor (through the UI not just code.) As many users have been taught specific work flows and will not know how to handle the brand new experience. Having a simple way to add back work flow tools/screens is very important.

EDIT.
It would be helpful to be able to tie in the Gutenberg experience user interface UX etc into the Styles/Header/Template views. At the moment it feels too removed from a familier Gutenberg experience. Having familiar UX elements clearly seen will help the user feel more comfortable.

**Comment by azhiya** [comment]
Whilst the accessibility team hasn't reviewed the prototype, we [agreed](https://wordpress.slack.com/archives/C02RP4X03/p1617381530408100) that this makes sense to us. We decided to review the prototype and comment if we see it being an issue.

**Comment by paaljoachim** [comment]
I am bringing up this issue during a design triage. As James and I had a discussion about this issue in the design channel. I am circling back to it. One thing that comes to mind is that the Appearance menu can contain various template areas and more.
Default to show would likely be Header and a Footer. Then based on what the user adds additional areas might show up. 

Here is a ticket where @OGlekler suggests adding Reusable blocks to the Appearance menu.
https://core.trac.wordpress.org/ticket/51534

One hesitation I have.
The Appearance menu can end up with a lot of different Template areas, Reusable blocks, Patterns etc. Making the list fairly long. Making it hard to have a good overview.

**Comment by jameskoster** [comment]
One thing Kelly and I didn't find a solution for yet was where to place general template parts. We had one idea to reuse the "Widgets" menu label since there is some conceptual overlap there – a widget is very similar to a general template part in its purpose. Another option could be to group them with top level templates somehow.

Personally I'm not a fan of putting the Reusable Blocks item in the Appearance menu. One of the defining characteristics of a Reusable Block versus something like a Template Part (or a pattern) is that its intended use is for displaying specific _content_. It's already easy for folks to confuse these two (three) features... and putting them in the same menu area probably isn't going to help matters.

**Comment by jameskoster** [comment]
Distilling this down in to actionable tasks:

- Remove top level "Site Editor" menu item from the wp-admin navigation
- If a single block template part assigned to the header `area` exists, add a "Header" menu item under Appearance which links to the isolated editing view for this template part.
- If _multiple_ block template parts exist that are assigned to the header `area` add a "Headers" menu item under Appearance which links to a filtered view of the template parts post type list that only displays Headers.
- Apply the same logic from the two previous points to footers.
- Update the "Templates" link under appearance to link to the mosaic view of templates (#20477) when built
- Add a "Styles" link under Appearance which links to the mosaic view of templates with the global styles sidebar opened
- Add a "Widgets" link under Appearance which links to a filtered view of the template parts post type list that only displays General template parts. "Widgets" label needs design :)

Unsure if these need to be _separate_ issues.

Overall this issue is currently blocked by #20477.

**Comment by carolinan** [comment]
I think repurposing the term widgets is a bad idea that will only cause more confusion.
If nothing else, consider how this will effect support when you are trying to describe and solve an issue.

**Comment by carolinan** [comment]
Also I feel strongly that these UX discussions should be public and not in private discussions. 
So when available please provide links to Figma and public slack discussions.

I am asking this because this does affect what issues and pull requests to prioritize, as any fixes to specific areas will be short term and may just be a waste of our time.

**Comment by kellychoffman** [comment]
> Also I feel strongly that these UX discussions should be public and not in private discussions.
> So when available please provide links to Figma and public slack discussions.

Agree. All links to Figma and discussions are in the comments above. One missing is a conversation that was started yesterday in Slack to discuss https://github.com/WordPress/gutenberg/issues/29630#issuecomment-814729489: https://wordpress.slack.com/archives/C02S78ZAL/p1617963102014400

**Comment by celloexpressions** [comment]
The features in FSE currently live together in a “site editor.” As #30496 explores, this interface is not a full replacement for the customizer yet. That issue proposes a few potential paths forward, each of which would impact the admin menu structure. A decision should likely be made on that issue before the menu approach is finalized.

There is discussion in #26012 about potentially consolidating the widgets and menus/navigation screens into the customizer. The “header” and “background” appearance sub-menus already function as deep links into the customizer (since 2014). Global styles would fit well into the customizer with its current sidebar-based UI. If template editing (the primary feature in the current site editor) also becomes accessible through the customizer (approach 2 proposed in #30496), then everything could live in a unified interface regardless of the level of theme support for blocks.

“Appearance” doesn’t seem like the right container for the site editing features. It implies styling options more than layout and template part content editing. Is there a better word? “Customize” might actually work—it implies optional and gradual editing from a default state (for both styles and templates) as a primary distinction from post and page content editing. We could end up with something like this in WP admin and in the customizer:

![wp-admin-new-menu](https://user-images.githubusercontent.com/7875961/114286848-4d7e9a00-9a17-11eb-9db2-fc6df1db787f.JPG)
![wp-customize-new-menu](https://user-images.githubusercontent.com/7875961/114286849-4e173080-9a17-11eb-9b07-5913d5931cb5.JPG)

**Comment by aristath** [comment]
My 2c would be to leave the customizer out of it...
Personal opinion follows:

The site-editor is not a replacement for the customizer, nor is global-styles. They are just different interfaces, for different scenarios and use-cases. The fact that we have a UI in the customizer for editing some options doesn't mean that everything should be in there... not when we now have a UI that fits better with the vision of where WP is headed.
The customizer had a good run. It served its purpose for years, but it feels inconsistent with where the ecosystem is heading.
It can still be used by classic themes, but there's no reason to force it in FSE; not when we have better options. Other than force-of-habit for existing users, there is no benefit to new users.
Things should be allowed to come to an end naturally and not dragged on for years impeding progress in other areas. At this point in time with FSE, the customizer feels more like a hindrance, something we need to take into account to make sure things don't break. It doesn't feel to me like a home worth building on.

**Comment by jameskoster** [comment]
> I think repurposing the term widgets is a bad idea that will only cause more confusion.
> If nothing else, consider how this will effect support when you are trying to describe and solve an issue.

That's a good point. 

I'm hesitant to throw another name in to the mix since we already have so many, but it is perhaps unavoidable. Even if we combined general template parts with top level (index etc) templates, a label would still be required in order to provide any kind of filtering.

Thinking semantically, something like "Template Sections" (or just "Sections") might work? 

> The HTML `<section>` element represents a generic standalone section of a document, which doesn't have a more specific semantic element to represent it.
> – [developer.mozilla.org](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/section)

In this case the "more specific semantic element[s]" would be Headers, Footer, Sidebars, and any other template part `area`s we add in the future. It helps that "Section" is a fairly user-friendly term in this context.

Other options:

* Segments 🍫
* Pieces 🧩
* Modules ⚙️
* Parts 🙈 

---

> “Appearance” doesn’t seem like the right container for the site editing features.

I think there may be a discussion to be had there, but probably in a separate issue once the changes here are implemented.

**Comment by carolinan** [comment]
I'd love to see different visual flow charts of how users would get from step A-Z. I think for me that would make it easier to see where we might run into problems with current and proposed flows.

**Comment by carolinan** [comment]
There are so many things to consider. There already is a header menu item that takes users to the customizer. What happens when a hybrid theme, that has support for both, is installed? Would there be two header menu items in different positions?

**Comment by aristath** [comment]
> There already is a header menu item that takes users to the customizer. What happens when a hybrid theme, that has support for both, is installed? Would there be two header menu items in different positions?

I assume it would depend on _what kind of hybrid theme_ it is... because the term hybrid can encompass lots of things.
If it is hybrid in the sense that it uses a block-based template for the header/footer as per https://github.com/WordPress/gutenberg/pull/30345, then it would take them to the template-part editor and not the customizer

**Comment by jameskoster** [comment]
> I'd love to see different visual flow charts of how users would get from step A-Z. 

Which flows in particular are you thinking of? I'm happy to work on more mockups :)

**Comment by carolinan** [comment]
All of them 😆 

I thought the new navigation screen was meant for classic, not block themes, but I see it in the screenshot above.

This is going to be another rant, but all I want is to understand better.

- The benefit of having isolated views of the header and footer template part is that the UI can be less overwhelming.
**Is this the main reason why header would need to be its own menu item?**

- The downside to having isolated views of the header and footer template parts is that it is more difficult to see what the full page/site will look like.

Does it make sense to call these template parts at all?
If these two are handled differently, then where do other template parts go?
As a theme developer I want to use template parts to make sure I don't need to repeat code. So, any area that may be the same on different templates. Sidebar, main (query goes here), post meta (author, date, category), comment area...

I wonder if Templates need to be it's own parent menu item not under appearance, site, etc. Maybe templates is the main menu and header, footer and template parts/areas are the sub menus.

**Comment by jameskoster** [comment]
> I thought the new navigation screen was meant for classic, not block themes, but I see it in the screenshot above.

I don't think that's the case? The navigation screen remains a place to manage all your navigation menus in one place. Whether those menus are block-based or not will depend on your theme etc.

> Is this the main reason why header would need to be its own menu item?

That is the primary reason, yes. We're also perpetuating a historical menu item. "Appearance > Header" has always taken you to a view where you can customise your site header. So going back to your documentation point from before there would appear to be some value in keeping this.

> If these two are handled differently, then where do other template parts go?

This is the discussion we had just a few comments above. "Widgets" was one option, others options are "Sections", etc. "Template Parts" is perhaps a bit too technical for the average user.

> templates is the main menu and header, footer and template parts/areas are the sub menus.

This is a perfect example of the sort of thing we'll be able to do with the drill-down menu system that already exists in the site editor. Until then I would argue that templates still fit inside "Appearance".

**Comment by pablohoneyhoney** [comment]
Full disclosure I'm not caught up in all related conversations in all channels, and perhaps I'm missing a bit of context on strenuous cases in a transition-to-block-based-themes world. 

From a more agnostic UX standpoint, my impression is that the taxonomy proposed is a bit complex to combine, terms have different means and semantic levels. It also provides a shallow interaction model (in and out) that I’m not sure it’s scalable. Even scaling to ideas like:

> Here is a ticket where @OGlekler suggests adding Reusable blocks to the Appearance menu.
> https://core.trac.wordpress.org/ticket/51534

which I don't fully agree with, but it does surface a scale challenge.

Related, it could also make sense to consider _Design_ as an umbrella term instead of _Appearance_, since this doesn’t really encompass what would happen within, in all various situations, current, desired, or in-transition. _Customize_ could make sense in English but it has a poor translation to Latin languages (_personalizzare_ in Italian). 

If we do however want to try the [proposal above](https://github.com/WordPress/gutenberg/issues/29630#issuecomment-811617222) (I’m aware it’s hard to compromise a perfect solution), it’s important we provide contextual validation of where you are and land (not fully clear and consistent in the prototype at the moment).
I was a bit confused with some details in the actual moments where you land (canvas bg, GS preview, the elevations, some have a top title but not others), but I believe that wasn’t the actual purpose of the prototype. 

Per this note by @azhiya (fair nevertheless):

> Whilst the accessibility team hasn't reviewed the prototype, we agreed that this makes sense to us. We decided to review the prototype and comment if we see it being an issue.

I'm troubled with an a11y assessment of the above flows as it can draw the wrong conclusions yet, given the details above noted and that the prototype didn't seem to aim at solving all the flows but finding an initial assumption on the right-enough navigation model. 

Kudos to the push on such an intricate challenge.

**Comment by talldan** [comment]
> I don't think that's the case? The navigation screen remains a place to manage all your navigation menus in one place. Whether those menus are block-based or not will depend on your theme etc.

The navigation screen (as in this project -  https://github.com/WordPress/gutenberg/projects/31) has been developed for classic (non block-based) themes. It doesn't have any FSE functionality.

I would assume the proposal is a way to edit a navigation block in a template part (or something similar) more directly than just editing it in a full template (ie header)? In that respect, it sounds very similar to the navigation screen, but then block-based themes could provide editor styles and support other more advanced features of the navigation block, so that would be a point of distinction.

**Comment by jameskoster** [comment]
Ah yes, some slightly crossed-wires there on my part. 

Re-iterating what Dan said: It may be that we have a dedicated menu management screen for FSE themes in the future (similar to template parts). But it is unlikely that will be the current format of the existing block-based navigation screen :) 

So going back to the original question, whether the Appearance > Navigation menu item appears would seem to be situational based on the type of theme that is active, and that is outside the scope of this particular issue.

**Comment by carolinan** [comment]
It seems like the easiest way is to let template parts be listed in a section of the template editor (not the current, but the view that does not yet exist and that replaces the site editor)
With the exception of the header link that leads directly to the single template part view, template parts has no menu item of its own.

Don't mind me, I am just a month behind everyone else ;)

**Comment by carolinan** [comment]
What does this mean for the documentation? If we call it template editor, this is confusing because there are actually 3 views today and two views for 5.8:

Site editor
Editing via the appearance > templates menu
Editing via post editor -> edit template link

The last two are not the same views, and I don't know what the plan is for them.

**Comment by jameskoster** [comment]
> It seems like the easiest way is to let template parts be listed in a section of the template editor (not the current, but the view that does not yet exist and that replaces the site editor)

The latest designs indicate that there won't be a single replacement for the Site Editor. We _could_ list general template parts on the Template Mosaic view, perhaps that's what you're referring to? :D 

They'd still need a label though :) 

> Editing via the appearance > templates menu
> Editing via post editor -> edit template link

> The last two are not the same views, and I don't know what the plan is for them.

Good question. My understanding from the [go/no go discussion](https://make.wordpress.org/core/2021/04/15/full-site-editing-go-no-go-april-14-2021/) is that only the Post Editor -> Edit Template flow will be in 5.8.

But my feeling is that the following flows should both take you to the same view:

* Appearance > Templates > Page Template
* Edit Page > Edit Template link > Page Template

**Comment by carolinan** [comment]
I like areas or parts, but then the label does not indicate their close relationship with templates.

**Comment by carolinan** [comment]
The problem with the Post Editor -> Edit Template flow  is that there is no way to delete user created templates. (And to edit a template you first need to find a post or page that uses that template...)

**Comment by jameskoster** [comment]
> The problem with the Post Editor -> Edit Template flow is that there is no way to delete user created templates.

Issue opened here: https://github.com/WordPress/gutenberg/issues/31076

**Comment by noisysocks** [comment]
I spoke with @kellychoffman and this is awaiting one final design iteration. @kevin940726 has graciously volunteered some of his time to do the dev once we land on a design.

**Comment by carolinan** [comment]
Is there a place where we can see the proposal?

**Comment by Mamaduka** [comment]
@carolinan I think these are the official posts about the proposal - [part 1](https://make.wordpress.org/design/2021/10/15/site-editing-ia-concepts-how-to-surface-and-access-new-features/) and [part 2](https://make.wordpress.org/design/2021/10/22/site-editing-ia-concepts-part-2/).

**Comment by kellychoffman** [comment]
Taking into account the feedback received and what was working from both concepts on both posts, proposing the following as an MVP: 

Click on Appearance and you see Editor (beta) menu item. This keeps the concept of updating your look and feel of your site within the Appearance menu item, where current users are used to going to Customize. It is also intuitive for new users and matches the iterative approach product development has taken.
![WP admin](https://user-images.githubusercontent.com/5835847/139321953-d6f07143-ef4a-45e8-9a8d-797be4f948f2.png)

From there, you are brought to the homepage of your site, where you can immediately start to edit it – whether its a static page or your latest posts. Example of the latter: 
![IF latest posts_ Brought to HP (templates) block sidebar open](https://user-images.githubusercontent.com/5835847/139322520-72571986-0aa8-414c-b3e7-a8170d000f9d.png)

From here, if you click on the W menu in the upper left corner, it opens a menu where you would be able to access Styles, Templates, and Template parts. Only including these three items would remove the duplication issue brought up in [this iteration](https://make.wordpress.org/design/2021/10/22/site-editing-ia-concepts-part-2/).
![Nav-template-parts](https://user-images.githubusercontent.com/5835847/139322761-cd7530f8-9257-4441-ab56-5700d00e938b.png)

Click on Styles to update the colors, typography, and layout of your site. This also shows what a welcome guide could look like, which could be useful for big new features.
![Styles](https://user-images.githubusercontent.com/5835847/139323057-8feda63b-6d38-4448-a9d4-87d212491e62.png)

If you open the W menu again and click on Templates, you’ll view a list of all the templates you can now edit visually. (Since this design is new and we are in a time crunch, we can fall back to the current wp-admin-style list view.)
![Templates](https://user-images.githubusercontent.com/5835847/139323111-468693c0-c7ce-4bc1-b6da-d2eabc480230.png)

Editing pages and posts will continue to be done where they exist today, as mentioned in the "Separate" concept, with the addition of a layout toggle within the Preview dropdown: (https://github.com/WordPress/gutenberg/issues/27847#issuecomment-765578813)

**Comment by carolinan** [comment]
And where is the theme file editor?
Perhaps under Tools?

**Comment by kellychoffman** [comment]
Tools could work. Also wondering if we could we keep it under Appearance - or would that be too confusing?

**Comment by noisysocks** [comment]
Thanks @kellychoffman! Let's at least get something rough merged by November 3 (Gutenberg 11.9 RC / WordPress 5.9 feature freeze). We can polish and fix bugs during the 5.9 beta period.

**Comment by carolinan** [comment]
@kevin940726 Can you list the follow ups mentioned in the issue?
Or did you mean the issue should not be closed until this is ported to core?

**Comment by kevin940726** [comment]
I mean the design in https://github.com/WordPress/gutenberg/issues/29630#issuecomment-954159446 has not been implemented yet. We can definitely open another issue for that though.

**Comment by noisysocks** [comment]
Are we able to implement the rest of the design by Friday?

**Comment by jameskoster** [comment]
> (Since this design is new and we are in a time crunch, we can fall back to the current wp-admin-style list view.)

Is it technically feasible to do this, but retain the Editor navigation? Something like:

<img width="1224" alt="Screenshot 2021-11-02 at 12 01 29" src="https://user-images.githubusercontent.com/846565/139842433-078adfbe-0799-4e74-8ca0-68ac91214012.png">

Otherwise the UX may feel awkward as you dip in and out of wp-admin.

I opened several issues and a couple of PRs relating to this view last week:

* https://github.com/WordPress/gutenberg/issues/35990
* https://github.com/WordPress/gutenberg/issues/35991
* https://github.com/WordPress/gutenberg/issues/35994
* https://github.com/WordPress/gutenberg/issues/35995
* https://github.com/WordPress/gutenberg/pull/35996
* https://github.com/WordPress/gutenberg/pull/35997

**Comment by kevin940726** [comment]
> If you open the W menu again and click on Templates, you’ll view a list of all the templates you can now edit visually. (Since this design is new and we are in a time crunch, we can fall back to the current wp-admin-style list view.) ![Templates](https://user-images.githubusercontent.com/5835847/139323111-468693c0-c7ce-4bc1-b6da-d2eabc480230.png)

I wonder why we want to keep the editor interface (the header with undo/redo buttons and the inspector etc) in this screen? Should this be a new path in the url, or should we implement it in a modal/popup style?

**Comment by jameskoster** [comment]
> I wonder why we want to keep the editor interface (the header with undo/redo buttons and the inspector etc) in this screen? 

Apologies, that is an oversight. Initially I think we should keep the top bar in this view very simple:

<img width="1262" alt="Screenshot 2021-11-03 at 09 52 31" src="https://user-images.githubusercontent.com/846565/140039726-0cd41cf0-d78d-47e1-82e3-6a5567fffaad.png">

The "Add new" button should expose a popover, effectively behaving the same as this affordance in the current site editor navigation:

<img width="531" alt="Screenshot 2021-11-03 at 09 54 39" src="https://user-images.githubusercontent.com/846565/140039909-249df089-eafa-4758-a7d0-31b794250b57.png">

> Should this be a new path in the url, or should we implement it in a modal/popup style?

Good question. Since this screen is accessed via navigation menu item, a unique url seems appropriate?

**Comment by kevin940726** [comment]
Makes sense to me! Do we have a preferred unique path for this screen yet? I feel like ideally we should make this into a _semi-SPA style_ editor, that is, use history API to navigate between screens. However, I don't think we have enough time before the beta though 😅 .

Even if we could fallback to _wp-admin-style list view_, I'm not so sure how that could be implemented along side the new W menu sidebar. (Maybe I'm just not experienced with this part though, could use some helps from PHP experts?)

What would be the absolute minimal that we should definitely do before the feature freeze? Leave it as is? Or should we implement the new W menu but the links point directly to the current wp_admin templates list (`/wp-admin/edit.php?post_type=wp_template`)? Given the limited time, I think we should do it step-by-step so that we can improve it gradually.

One more question: what does the "Styles" menu item suppose to link to? Should it just open the Styles sidebar? What if the Styles sidebar is already opened? What should it do when the users are in the templates(-parts) list view?

**Comment by talldan** [comment]
BTW, there's a plan to add a 'Navigation Menus' menu item under Appearance in https://github.com/WordPress/gutenberg/pull/36126.

It wouldn't be possible to edit these menus right away, but a way to manage them (rename, delete) is still valuable.

**Comment by jameskoster** [comment]
More good questions @kevin940726.

Falling back to the wp-admin templates list view feels like a poor navigational experience to me. I put a prototype together [here](https://www.figma.com/proto/LZiM9diybGa4WKM44zRcIq/Appearance-Menu?page-id=3450%3A128611&node-id=3458%3A127656&viewport=326%2C48%2C0.42&scaling=scale-down&starting-point-node-id=3458%3A127656) to test it and would be interested to hear feedback on that.

Handling the Styles menu item is tricky. One option would be to have that menu item behave identically to the toggle in the top bar. Whatever we do here is going to feel sub-optimal to some extent since the interaction isn't navigational despite its position in the UI suggesting that it is. You can try this in the prototype above as well.

---

These two quirks make me wonder if an even simpler MVP would be to simply remove the navigation in the site editor altogether to begin with, and put links to Templates and Styles in the Appearance menu. I made a [prototype](https://www.figma.com/proto/LZiM9diybGa4WKM44zRcIq/Appearance-Menu?page-id=3450%3A126841&node-id=3450%3A126842&viewport=326%2C48%2C0.12&scaling=scale-down&starting-point-node-id=3450%3A128168) for this one too.

**Comment by kevin940726** [comment]
> These two quirks make me wonder if an even simpler MVP would be to simply remove the navigation in the site editor altogether to begin with, and put links to Templates and Styles in the Appearance menu. I made a [prototype](https://www.figma.com/proto/LZiM9diybGa4WKM44zRcIq/Appearance-Menu?page-id=3450%3A126841&node-id=3450%3A126842&viewport=326%2C48%2C0.12&scaling=scale-down&starting-point-node-id=3450%3A128168) for this one too.

I personally like this more as a MVP, the UX feels more natural than the others. We can implement this first, and take our time implementing the new screen in https://github.com/WordPress/gutenberg/issues/29630#issuecomment-954159446 later. Not sure if the new screen would make it to 5.9 though. WDYT?

**Comment by jameskoster** [comment]
I think that makes sense, but would like to hear @kellychoffman's input.

**Comment by kellychoffman** [comment]
My take: Let's move forward with this option.

I think it comes with some trade-offs that are inherent with the current wp-admin nav menu, but it feels like the best option that will allow us to move forward and provides a base to work from.

We can always approach introducing the W menu in future releases if it makes sense. It will give us time to rethink and rework the nav overall.

**Comment by jameskoster** [comment]
With https://github.com/WordPress/gutenberg/pull/36064 merged I _think_ we can close this particular issue.

There's still lots to do of course, but we've opened other issues that will execute the design outlined in https://github.com/WordPress/gutenberg/issues/29630#issuecomment-954159446 (and subsequent discussion).

**Comment by kellychoffman** [comment]
@jameskoster: In light of [PR #36379](https://github.com/WordPress/gutenberg/pull/36379), and the confusing nature of the nav structure as is (see image below), I'm wondering if we should reconsider something closer to this approach: https://github.com/WordPress/gutenberg/issues/29630#issuecomment-954159446. (The reason we originally removed the W menu, was because this was [not possible](https://github.com/WordPress/gutenberg/issues/29630#issuecomment-958874048).)

Overwhelming, confusing nav structure:
![image (21)](https://user-images.githubusercontent.com/5835847/142083060-8023d8e5-c7c1-4978-8dc9-8f32e76557c1.png)

**Comment by jameskoster** [comment]
Certainly, when https://github.com/WordPress/gutenberg/pull/36379 is merged the Styles, Templates, and Template Parts links move from the Appearance menu in wp-admin to the W menu in the editor. 

This was always the North Star, but it wasn't clear whether or not we could get it done in time for 5.9.
